eng/native/gen-buildsys.sh

```diff
@@ -106,6 +106,7 @@ if [[ "$host_arch" == "wasm" ]]; then
     fi
 fi
 
+echo "$cmake_command --no-warn-unused-cli -G \"$generator\" \"-DCMAKE_BUILD_TYPE=$buildtype\" \"-DCMAKE_INSTALL_PREFIX=$__CMakeBinDir\" $cmake_extra_defines $__UnprocessedCMakeArgs \"${cmake_extra_defines_wasm[@]}\" -S \"$1\" -B \"$2\""
```

Comment:
We are using this pattern to deduplicate command echo-then-eval: https://github.com/dotnet/runtime/blob/0205063ee6f87c4ac3bc22a48b71d930a03cbe5b/eng/native/build-commons.sh#L174-L176 keep it consistent?